### ISSUE TITLE
Inf.Rat == Rational bug

### ISSUE THREAD

**Issue by b2gills** [rakudo/rakudo#2578]
[following]: https://github.com/rakudo/rakudo/blob/4a968059c2ea8e7ea4cd1f0b96938f5019d6b6cb/src/core/Rat.pm6#L222-L230

## The Problem

The `&infix:<==>(Rational:D \a, Rational:D \b)` candidate doesn't handle `Inf.Rat` properly.

```perl6
1/0 == 1/3; # True
1/3 == 1/0; # True
```

Since both `Inf.Rat` and `NaN.Rat` have zero for a denominator, the [following][] gets to the code marked with `# NaN != NaN`.

```perl6
multi sub infix:<==>(Rational:D \a, Rational:D \b) {
    nqp::hllbool(
      nqp::isfalse(a.denominator) || nqp::isfalse(b.denominator)
        ?? nqp::iseq_I(a.numerator, b.numerator)
          && nqp::istrue(a.numerator) # NaN != NaN
        !! nqp::iseq_I(
            nqp::mul_I(a.numerator, b.denominator, Int),
            nqp::mul_I(b.numerator, a.denominator, Int)))
}
```

**Comment by lizmat** [comment]
Fixed with c11e7e6bcd .

**Comment by lizmat** [comment]
Tests added with https://github.com/perl6/roast/commit/6feffac83c
